examples/full_tutorial.ipynb

```diff
@@ -0,0 +1,882 @@
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "38e8afeb-274c-4c62-99a7-6200fc9c179d",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from IPython.display import HTML\n",
+    "css_rules = \"\"\"\n",
+    "button {\n",
+    "    background-color: yellow;\n",
+    "    border-radius: 10px;\n",
+    "}\n",
+    "\"\"\"\n",
+    "HTML(\"<style>\" + css_rules + \"</style>\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cb19480f-b3c6-44be-968c-06edf944d42f",
+   "metadata": {},
+   "source": [
+    "You can also load CSS from a file.  \n",
+    "And you can add extra CSS without overwriting the previous CSS import."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c4a644e0-1dbe-4f50-9bb5-5a54f1fedbaf",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from IPython.display import HTML\n",
+    "from pathlib import Path\n",
+    "css_rules = Path(\"styles_orange.css\").read_text()\n",
+    "HTML(\"<style>\" + css_rules + \"</style>\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f571eadd-1a94-4ffe-af43-1c5ef1ec2498",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Note: When you clear the output of the previous cell, this CSS style will be removed from the notebook."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "b50fbd27-5679-48ea-b2cc-ff9c5c79f826",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Next, we want to **add parameters to the widget.**  \n",
+    "First naive approach: Using f-strings.\n",
+    "That works, but is bad for two reasons:\n",
+    "1. curly brackets from f-strings will collide with the curly brackets from css. \n",
+    "Using curly brackets in an f-string as a normal character can be done by doubling them like this : `{`to `{{`.\n",
+    "2. It is not possible to pass this parameter into the `MyExampleWidget` class, as it does not have an init function.\n",
+    "TODO: give a reason why there is no normal init function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6fdda6ec-d6df-42bb-964c-66b686596ff2",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import ipyreact\n",
+    "\n",
+    "# ❌❌❌  WARNING: THIS CODEBLOCK IS NOT GOOD PRACTICE ❌❌❌ \n",
+    "class MyExampleWidget(ipyreact.ReactWidget):\n",
+    "    my_message = \"Hello World\"\n",
+    "    _esm = f\"\"\"\n",
+    "    import * as React from \"react\";\n",
+    "\n",
+    "    export default function MyButton() {{\n",
+    "        return <button> {my_message} </button> \n",
+    "        }};\"\"\"\n",
+    "MyExampleWidget()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "944363e3-92b1-4ee6-8e16-1a2f6879fefb",
+   "metadata": {},
+   "source": [
+    "Instead, we can **use traitlets.**  \n",
+    "traitlets are variables that can be used both in JavaScript and python.  \n",
```

Comment:
Traitlets that are tagged with `sync=True` will be available not only on the Python side, but also in the frontend.